# HG changeset patch
# User Dieter Baron <dillo@nih.at>
# Date 1367058218 -7200
# Node ID a94e409bc55a76b02705dfc64ea04abaf32f64c2
# Parent  cfbac9adc0780ef3f301b68064dc3e7690eca8c4
Initialize more members in zip_source_zip, so zip_source_stat works before zip_source_open (needed by zip_close).

diff -r cfbac9adc078 -r a94e409bc55a lib/zip_source_crc.c
--- a/lib/zip_source_crc.c	Sat Apr 27 12:22:33 2013 +0200
+++ b/lib/zip_source_crc.c	Sat Apr 27 12:23:38 2013 +0200
@@ -66,8 +66,12 @@
 	return NULL;
     }
 
+    ctx->eof = 0;
     ctx->validate = validate;
-
+    ctx->e[0] = ctx->e[1] = 0;
+    ctx->size = 0;
+    ctx->crc = 0;
+    
     return zip_source_layered(za, src, crc_read, ctx);
 }
 
